airflow/providers/amazon/aws/sensors/s3_upload_session_complete.py

```diff
@@ -0,0 +1,171 @@
+        has passed with no additional objects sensed.
+    :type inactivity_period: float
+    :param min_objects: The minimum number of objects needed for upload session
+        to be considered valid.
+    :type min_objects: int
+    :param previous_objects: The set of object ids found during the last poke.
+    :type previous_objects: set[str]
+    :param allow_delete: Should this sensor consider objects being deleted
+        between pokes valid behavior. If true a warning message will be logged
+        when this happens. If false an error will be raised.
+    :type allow_delete: bool
+    """
+
+    template_fields = ('bucket_name', 'prefix')
+
+    @apply_defaults
+    def __init__(self,
+                 bucket_name: str,
+                 prefix: str,
+                 aws_conn_id: str = 'aws_default',
+                 verify: Optional[Union[bool, str]] = None,
+                 inactivity_period: float = 60 * 60,
+                 min_objects: int = 1,
+                 previous_objects: Optional[Set[str]] = None,
+                 allow_delete: bool = True,
+                 *args, **kwargs) -> None:
+
+        super().__init__(*args, **kwargs)
+
+        self.bucket = bucket_name
+        self.prefix = prefix
+        if inactivity_period < 0:
+            raise ValueError("inactivity_period must be non-negative")
+        self.inactivity_period = inactivity_period
+        self.min_objects = min_objects
+        self.previous_objects = previous_objects if previous_objects else set()
+        self.inactivity_seconds = 0
+        self.allow_delete = allow_delete
+        self.aws_conn_id = aws_conn_id
+        self.verify = verify
+        self.last_activity_time: Optional[datetime] = None
+        self.hook = None
+
+    def _get_aws_hook(self):
+        if not self.hook:
+            self.hook = S3Hook(aws_conn_id=self.aws_conn_id, verify=self.verify)
+        return self.hook
+
+    def is_bucket_updated(self, current_objects: Set[str]) -> bool:
+        """
+        Checks whether new objects have been uploaded and the inactivity_period
+        has passed and updates the state of the sensor accordingly.
+
+        :param current_objects: set of object ids in bucket during last poke.
+        :type current_objects: set[str]
+        """
+        current_num_objects = len(current_objects)
+        if current_objects > self.previous_objects:
+            # When new objects arrived, reset the inactivity_seconds
+            # and update previous_objects for the next poke.
+            self.log.info("New objects found at %s resetting last_activity_time.",
+                          os.path.join(self.bucket, self.prefix))
+            self.log.debug("New objects: %s",
+                           "\n".join(current_objects - self.previous_objects))
+            self.last_activity_time = datetime.now()
+            self.inactivity_seconds = 0
+            self.previous_objects = current_objects
+            return False
```

Comment:
This method is called `is_bucket_updated`, and it has been here, but we are returning False -- Given the method name, I would expect this to be True here.

Reply:
(You can either change the return value, or change the name)